### DISCUSSION TITLE
Custom Action Buttons while printing

### DISCUSSION THREAD

**Discussion by M1dn1ghtN1nj4** [UnchartedBull/OctoDash#1660] [Ideas]
Unless I missed something, this would be a great addition.  I have a NeoPixel setup on my printer, and have configured two of the 6 action buttons for it (one for on, one for off).  Unfortunately, that only works when the printer is NOT printing.  I'd like to be able to turn the NeoPixels on and off mid print, without having to rely on my hardwired switch to do it.

If I do it with the switch, there is no way to turn them back on unless I log into OctoPrint, go to terminal, and issue another M150 command.

I thought I saw a way to open the 6 action buttons from a right side pull out menu somewhere.  Might have been dreaming.  lol

But yeah, being able to control custom actions like that mid-print would be a huge pull.  Maybe even have a full screen worth of buttons to work with (like 3 rows, 4 or 5 columns).

**Comment by UnchartedBull** [comment]
#828. Most of the pre-work is done (v3.0.0 will have custom actions on sleep screen as you probably already have noticed). Will be released with 3.0.x or 3.1.0.

**Comment by M1dn1ghtN1nj4** [comment]
I didn't see that.  Was that in the finished category for v3?  If it is I
totally missed it.

I'm still building v2.1.2 from current main branch.  Latest build working
fine.  Still had the initial hiccup trying to connect to OctoPrint, but a
couple reboots later it worked.

**Reply by UnchartedBull** [reply]
The issue for custom actions on sleep screen was in v3 completed, custom actions on print screen got postponed to 3.0.x or 3.1.0.
